components/homme/src/theta-l_kokkos/cxx/HyperviscosityFunctorImpl.cpp

```diff
@@ -65,13 +73,41 @@ void HyperviscosityFunctorImpl::init_params(const SimulationParams& params)
     ExecViewManaged<Scalar[NUM_LEV]>::HostMirror h_nu_scale_top;
     h_nu_scale_top = Kokkos::create_mirror_view(m_nu_scale_top);
 
-    Kokkos::Array<Real,NUM_BIHARMONIC_PHYSICAL_LEVELS> lev_nu_scale_top = { 4.0, 2.0, 1.0 };
-    for (int phys_lev=0; phys_lev<NUM_BIHARMONIC_PHYSICAL_LEVELS; ++phys_lev) {
+    const auto etai_h = Kokkos::create_mirror_view(m_hvcoord.etai);
+    const auto etam_h = Kokkos::create_mirror_view(m_hvcoord.etam);
+    Kokkos::deep_copy(etai_h, m_hvcoord.etai);
+    Kokkos::deep_copy(etam_h, m_hvcoord.etam);
+
+    for (int phys_lev=0; phys_lev < NUM_LEV*VECTOR_SIZE; ++phys_lev) {
       const int ilev = phys_lev / VECTOR_SIZE;
       const int ivec = phys_lev % VECTOR_SIZE;
-      h_nu_scale_top(ilev)[ivec] = lev_nu_scale_top[phys_lev]*m_data.nu_top;
+
+      Real ptop_over_press;
+
+      //prevent padding of nu_scale to get nans to avoid last interface levels
+      //of w, phi to be nans, too
+      if( phys_lev < NUM_PHYSICAL_LEV ){
+        //etai is num_interface_lev, that is, 129 or 73
+        //etam is num_lev, so packs
+        if ( etai_h(0) == 0.0) {
+          ptop_over_press = etam_h(0)[0] / etam_h(ilev)[ivec];
+        }else{
+          ptop_over_press = etai_h(0) / etam_h(ilev)[ivec];
+        }
+      }else{
+          ptop_over_press = 0.0;
+      }
+
+      auto val = 16.0*ptop_over_press*ptop_over_press / (ptop_over_press*ptop_over_press + 1);
```

Comment:
Maybe it could help to add a comment that explains this line? Like, where does 16.0 come from (deduced from some params, or set after trial/error), or where is the x^2/ (x^2+1) (x=p0/p) coming from (e.g., deduced from some principle/formula, or an arbitrary fcn that has certain props).